src/utils/depo_client.ts

```
@@ -0,0 +1,121 @@
+    const descriptionBuffer = Buffer.alloc(200)
+    descriptionBuffer.write(description)
+
+    try {
+      if (this.wallet.publicKey) {
+        await this.program.methods.createEscrow(
+          Array.from(escrowId),
+          nameBuffer,
+          descriptionBuffer
+        ).accounts({
+          escrow: escrowKey,
+          signer: this.wallet.publicKey!,
+          systemProgram: SystemProgram.programId,
+        } as any)
+        .rpc()
+  
+        const escrowAccount = await this.program.account.escrow.fetch(escrowKey);
+        const escrow = new Escrow(escrowAccount);
+        console.log("Escrow:", escrow);
+        return escrow;
+      } else {
+        throw new Error("Wallet is not available");
+      }
+    } catch (error) {
+      console.error("Error creating escrow:", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Get an escrow by its UUID
+   * @param uuid - The UUID of the escrow
+   * @returns The Escrow object or null if not found
+   */
+  async getEscrow(uuid: string) {
+    try {
+      const id = uuid.replace(/-/g, '')
+      const escrowKey = this.getPdaKey(id)
+    
+      // First check if account exists to avoid the error
+      const accountInfo = await this.program.provider.connection.getAccountInfo(escrowKey);
+      if (!accountInfo) {
+        return null;
+      }
+      
+      const escrowAccount = await this.program.account.escrow.fetch(escrowKey);
+      const escrow = new Escrow(escrowAccount);
+
+      // TODO: Fetch Recipients & Depositors
+      // TODO: Fetch Conditions modules
+
+      return escrow;
+    } catch (error: any) {
+      return null;
+    }
+  }
+
+  /**
+   * Get all escrows
+   * @returns The list of Escrow objects
+   */
+  async getAllEscrows() {
+    try {
+      const escrows = await this.program.account.escrow.all();
+      return escrows.map(escrow => new Escrow(escrow.account));
+    } catch (error: any) {
+      console.error("Error fetching escrows:", error);
+      return [];
```

I am not sure about this pattern. I think it would be cleaner to return an error to the user ?

---

Returning an empty array gives the impression that "there are no escrows," when in reality, the fetch may have failed.

It would be better to propagate the error ?

---

Maybe, we can use toaster on that as well for the user

---

On that matter, toaster is working fine. I've just tested it on the validation of inputs condition in front.